src/util/tmpl/fd_treap.c

```diff
@@ -492,18 +493,20 @@ TREAP_(ele_query_const)( TREAP_(t) const * treap,
   return fd_ptr_if( !TREAP_IDX_IS_NULL( i ), ele + i, NULL );
 }
 
-static inline void
+static inline TREAP_(t) *
 TREAP_(ele_insert)( TREAP_(t) * treap,
                     TREAP_T *   e,
                     TREAP_T *   ele ) {
   TREAP_(idx_insert)( treap, (ulong)(e-ele), ele );
+  return treap;
 }
 
-static inline void
+static inline TREAP_(t) *
 TREAP_(ele_remove)( TREAP_(t) * treap,
                     TREAP_T *   e,
                     TREAP_T *   ele ) {
   TREAP_(idx_remove)( treap, (ulong)(e-ele), ele );
+  return treap;
```

Comment:
These methods in general and the seed function in particular seem to assume the elements come from a flat array. Is this mean to be used with fd_pool? If so, perhaps this should be explicit.

Reply:
See global comment below.